src/components/hostURL/HostURL.tsx

```diff
@@ -29,24 +29,44 @@ export default class HostURLConfiguration extends Component<HostURLConfigProps,
     }
 
     componentDidMount() {
-        getHostURLConfiguration().then((response) => {
-            let form = response.result;
-            if (!form) {
-                form = {
-                    id: undefined,
-                    key: "url",
-                    value: "",
-                    active: true,
-                }
-            }
-            this.setState({
-                view: ViewType.FORM,
-                form: form
-            })
-        }).catch((error) => {
-            showError(error);
-            this.setState({ view: ViewType.ERROR, statusCode: error.code });
-        })
+      let payload = {
+          ...this.state.form,
+          value: window.location.origin,
+      }
+      saveHostURLConfiguration(payload)
```

Comment:
If we are saving the host config and getting form in response, why do we need to call getHostURLConfiguration explicitly. can't we set the same value?